CMakeLists.txt

```diff
@@ -140,7 +123,34 @@ add_option(
     "no"
     "yes;no")
 add_option("BUILD_TEST_APPS" "Build the wolfBoot test apps (default: disabled)" "no" "yes;no")
-add_option("PYTHON_KEYTOOLS" "Use wolfCrypt-py for key generation and signing (default: disabled)" "no" "yes;no")
+add_option("PYTHON_KEYTOOLS" "Use wolfCrypt-py for key generation and signing (default: disabled)"
+           "no" "yes;no")
+add_option(
+    "PULL_LINKER_DEFINES"
+    "Pull partition addresses from the linker script instead of defining fixed addresses in target.h (default: disabled)"
+    "no"
+    "yes;no")
+
+# If PULL_LINKER_DEFINES is set, partition addresses and size info will come from the linker script,
+# so no need to pass them on the command line. These variables do need to be set for building the
+# test apps.
+if(NOT DEFINED PULL_LINKER_DEFINES AND NOT DEFINED BUILD_TEST_APPS)
```

Comment:
are there any other cases besides building the test apps where we need these to be checked?